### DISCUSSION TITLE
Is semaphore lock is required in dlt_initialize_socket_connection()

### DISCUSSION THREAD

**Discussion by Somesh55** [COVESA/dlt-daemon#722] [Q&A]
Hello Team,

I wanted to confirm is semaphore lock/unlock is required in dlt_initialize_socket_connection() function.

In our production environment, multiple apps are started at the boot time which are using DLT logs. And at this time dlt-daemon is not running, it is started after sometime ~20sec since boot. So, multiple threads are calling dlt_initialize_socket_connection() function and acquire the lock. And other threads are waiting to enter in to critical section.
Because of this waiting time, we are experiencing delay within our apps.

So, we are thinking to remove the semaphore lock in dlt_initialize_socket_connection() function if it is not necessary.

Could you please let us know the impact if we remove the semaphore lock in dlt_initialize_socket_connection()?
It would be great if you can share your feedback.

Thank you very much!

**Comment by minminlittleshrimp** [comment]
I think we need to look into how finite state machine has been designed in dltlib so that we can consider the need of locking here and there, locking in libdlt might not be easy for modification
